packages/tinybird/src/validation.ts

```diff
@@ -0,0 +1,81 @@
+import * as z from "zod";
+
+/**
+ * All available Vercel (AWS) regions
+ */
+export const availableRegions = [
+  "arn1",
+  "bom1",
+  "cdg1",
+  "cle1",
+  "cpt1",
+  "dub1",
+  "ewr1",
+  "fra1",
+  "gru1",
+  "hkg1",
+  "hnd1",
+  "icn1",
+  "kix1",
+  "lhr1",
+  "pdx1",
+  "sfo1",
+  "sin1",
+  "syd1",
+] as const;
+
+/**
+ * Values for the datasource ping_response_v2
+ */
+export const tbIngestPingResponse = z.object({
+  id: z.string(),
+  workspaceId: z.string(),
+  pageId: z.string(),
+  monitorId: z.string(),
+  timestamp: z.number().int(),
+  statusCode: z.number().int(),
+  latency: z.number().int(), // in ms
+  cronTimestamp: z.number().int().optional().default(Date.now()),
+  url: z.string().url(),
+  metadata: z
+    .record(z.string(), z.unknown())
+    .default({})
+    .transform((t) => JSON.stringify(t))
+    .optional(),
```

Comment:
Possible to pass a metadata as object and store it as string. And when retrieving it, transforming it back to JSON. This will be useful for the `Response` for example.